File: src/vibe_kanban_clone/schemas/task_run.py
"""Task run schemas."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class TaskRunRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    task_id: int
    model_id: int | None
    agent_id: int | None
    status: str
    prompt: str | None
    output: str | None
    usage: dict | None
    started_at: datetime | None
    finished_at: datetime | None
    error: str | None

File: server/src/schemas/taskRun.ts
import { z } from "zod";

export const TaskRunRead = z.object({
  id: z.number(),
  task_id: z.number(),
  model_id: z.number().nullable(),
  agent_id: z.number().nullable(),
  status: z.string(),
  prompt: z.string().nullable(),
  output: z.string().nullable(),
  usage: z.string().nullable(),
  started_at: z.string().or(z.date()).nullable(),
  finished_at: z.string().or(z.date()).nullable(),
  error: z.string().nullable(),
});

export type TaskRunRead = z.infer<typeof TaskRunRead>;
